src/knowledge_commitment/knowledge_commitment.hpp

```diff
@@ -60,8 +60,8 @@ struct knowledge_commitment {
 template<typename T1, typename T2, mp_size_t m>
 knowledge_commitment<T1,T2> operator*(const libff::bigint<m> &lhs, const knowledge_commitment<T1,T2> &rhs);
 
-template<typename T1, typename T2, mp_size_t m, const libff::bigint<m> &modulus_p>
-knowledge_commitment<T1,T2> operator*(const libff::Fp_model<m, modulus_p> &lhs, const knowledge_commitment<T1,T2> &rhs);
+template<typename T1, typename T2, typename T>
```

Comment:
This less-specific templatization loses information, and could cause future problems overloading `operator*` for other types.
Does the original exceed C++11, or is clang (and which version?) showing incomplete C+1+11 support here?